api/krusty/helmchartinflationgenerator_test.go

```diff
@@ -233,3 +233,283 @@ spec:
   type: ClusterIP
 `)
 }
+
+func TestHelmChartInflationGeneratorWithNamespace(t *testing.T) {
```

Comment:
Regarding [my question](https://github.com/kubernetes-sigs/kustomize/pull/4716#issuecomment-1251604604) about the potential conflict with kustomize's own `namespace` transformer. Can you add another unit test(s) to cover the following cases:
###  `namespace` configured by both kustomize and helm 
```yaml
# kustomization.yaml  
namespace: dev1 # managed-by-kustomize
helmChartInflationGenerator:
- chartName: nats
  chartVersion: v0.13.1
  chartRepoUrl: https://nats-io.github.io/k8s/helm/charts
  releaseName: nats
  releaseNamespace: dev2 # managed-by-helm
```
###  Base helm `namespace`, overlay kustomize `namespace`
@KnVerey Shall kustomize support this case? 
```yaml
# base/kustomization.yaml  
helmChartInflationGenerator:
- chartName: nats
  chartVersion: v0.13.1
  chartRepoUrl: https://nats-io.github.io/k8s/helm/charts
  releaseName: nats
  releaseNamespace: base # managed-by-helm
```
```yaml
# dev/kustomization.yaml  
resources: ../base
namespace: dev # managed-by-kustomize
```
###  Base kustomize `namespace`, overlay helm `namespace`
@KnVerey Shall kustomize support this case? 
```yaml
# base/kustomization.yaml  
namespace: base # managed-by-kustomize
```
```yaml
# dev/kustomization.yaml  
resources: ../base
helmChartInflationGenerator:
- chartName: nats
  chartVersion: v0.13.1
  chartRepoUrl: https://nats-io.github.io/k8s/helm/charts
  releaseName: nats
  releaseNamespace: dev # managed-by-helm
```